source/server/options_impl.cc

```diff
@@ -249,11 +252,55 @@ void OptionsImpl::parseComponentLogLevels(const std::string& component_log_level
   }
 }
 
+uint32_t OptionsImpl::count() const { return count_; }
+
 void OptionsImpl::logError(const std::string& error) const {
   std::cerr << error << std::endl;
   throw MalformedArgvException(error);
 }
 
+Server::CommandLineOptionsPtr OptionsImpl::toCommandLineOptions() const {
+  Server::CommandLineOptionsPtr command_line_options =
+      std::make_unique<envoy::admin::v2alpha::CommandLineOptions>();
+  command_line_options->set_base_id(baseId());
+  command_line_options->set_concurrency(concurrency());
+  command_line_options->set_config_path(configPath());
+  command_line_options->set_config_yaml(configYaml());
+  command_line_options->set_allow_unknown_fields(allow_unknown_fields_);
+  command_line_options->set_admin_address_path(adminAddressPath());
+  command_line_options->set_component_log_level(component_log_level_str_);
+  command_line_options->set_log_level(spdlog::level::to_c_str(logLevel()));
+  command_line_options->set_log_format(logFormat());
+  command_line_options->set_log_path(logPath());
+  command_line_options->set_service_cluster(serviceClusterName());
+  command_line_options->set_service_node(serviceNodeName());
+  command_line_options->set_service_zone(serviceZone());
+  if (mode() == Server::Mode::Serve) {
+    command_line_options->set_mode(envoy::admin::v2alpha::CommandLineOptions::Serve);
+  } else if (mode() == Server::Mode::Validate) {
+    command_line_options->set_mode(envoy::admin::v2alpha::CommandLineOptions::Validate);
+  } else {
+    command_line_options->set_mode(envoy::admin::v2alpha::CommandLineOptions::InitOnly);
+  }
+  if (localAddressIpVersion() == Network::Address::IpVersion::v4) {
+    command_line_options->set_local_address_ip_version(
+        envoy::admin::v2alpha::CommandLineOptions::v4);
+  } else {
+    command_line_options->set_local_address_ip_version(
+        envoy::admin::v2alpha::CommandLineOptions::v6);
+  }
+  command_line_options->mutable_file_flush_interval()->set_seconds(fileFlushIntervalMsec().count() /
```

Comment:
Do we have any setter macros/functions that make this type of conversion less error prone? We must be doing this somewhere? Take a look at the gRPC access log code maybe?